src/D2L.CodeStyle.Analyzers/CommonFixes/AddAttributeCodeFix.cs

```diff
@@ -46,9 +48,16 @@ CodeFixContext context
 				.ConfigureAwait( false ) as CompilationUnitSyntax;
 
 			foreach( var diagnostic in context.Diagnostics ) {
-				var identifierSpan = diagnostic.Location.SourceSpan;
+				// Sometimes the diagnostic is on the declaration that needs the attribute,
+				// if not it is the last "additional location" by convention.
+				var location = diagnostic.AdditionalLocations?.LastOrDefault() ?? diagnostic.Location;

```

Comment:
Thing of it like the list is `[diagnostic.Location] + diagnostic.AdditionalLocations` and we're picking the last element out of that list.

Some of the blocking diags are on method invocations, but we want to put the attribute up on the containing method definition, which we put some effort into locating originally in the analyzer.

This code fix also gets used to add [Immutable] onto types.